keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/jackson/SecurityGroupSpecDeserializer.kt

```diff
@@ -0,0 +1,34 @@
+package com.netflix.spinnaker.keel.ec2.jackson
+
+import com.fasterxml.jackson.databind.DeserializationContext
+import com.fasterxml.jackson.databind.InjectableValues.Std
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.deser.std.StdNodeBasedDeserializer
+import com.fasterxml.jackson.module.kotlin.convertValue
+import com.netflix.spinnaker.keel.api.Moniker
+import com.netflix.spinnaker.keel.api.ec2.SecurityGroupSpec
+import com.netflix.spinnaker.keel.core.name
+
+class SecurityGroupSpecDeserializer : StdNodeBasedDeserializer<SecurityGroupSpec>(SecurityGroupSpec::class.java) {
+  override fun convert(root: JsonNode, context: DeserializationContext) =
+    with(context.parser.codec as ObjectMapper) {
+      val moniker: Moniker = convertValue(root.path("moniker"))
+      SecurityGroupSpec(
+        moniker = moniker,
+        locations = convertValue(root.path("locations"))
+          ?: context.findInjectableValue("locations"),
+        description = root.get("description")?.textValue(),
+        inboundRules = copy().run {
+          injectableValues = Std(mapOf("name" to moniker.name))
+          root.get("inboundRules")?.let { convertValue(it) } ?: emptySet()
+        },
```

Comment:
This creates a unique mapper for the sub-tree with an injectable value based on the moniker we just parsed